File: library/src/main/java/com/hugocastelani/waterfalltoolbar/library/annotation/Dp.java
package com.hugocastelani.waterfalltoolbar.library.annotation;

import android.support.annotation.Dimension;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.LOCAL_VARIABLE;
import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.RetentionPolicy.CLASS;

/**
 * Created by Hugo Castelani
 * Date: 19/09/17
 * Time: 20:48
 *
 * Denotes that an integer parameter, field or method return value is expected
 * to represent a dp dimension.
 */
@Retention(CLASS)
@Target({METHOD, PARAMETER, FIELD, LOCAL_VARIABLE})
@Dimension(unit = Dimension.DP)
public @interface Dp {
}

File: library/src/main/java/com/hugocastelani/waterfalltoolbar/domain/Dp.kt
package com.hugocastelani.waterfalltoolbar.domain

/**
 * Created by Hugo Castelani
 * Date: 23/12/17
 * Time: 22:12
 */

data class Dp(var value: Float, private val density: Float) {
    fun toPixel(): Pixel {
        return Pixel((value * density + 0.5f).toInt(), density)
    }
}